### DISCUSSION TITLE
How to set hypothesis testing parameters to get more accurate conclusions?

### DISCUSSION THREAD

**Discussion by fqss0436** [UPPAALModelChecker/UPPAAL-Meta#239] [Q&A]
1) Bug: In the SMC parameter GUI,   "false negative(α)" is wrong, it is actually false  positive. "false positive(β)" as similarly.
2) Question1:How to set hypothesis testing parameters(+δ,-δ,α,β) to get more accurate conclusions?What is the significance of parameters(+δ,-δ) ? Why cannot these parameter be set to 0, and why must indifference region exist? 
3) Question2:In Query Result window, for example, (17244/117236  runs) H1:Pr()<=XX(prove H1 valid), if the hypothesis testing parameters(+δ,-δ,α,β)  changes,is it possiable that the same (17244/117236  runs) get different result  (prove H0 valid)?

**Comment by mikucionisaau** [comment]
1. Thank you for pointing this out!
2. When the outcomes fall into indifference region, it means that the outcomes are so close that we cannot distinguish them (within the range of statistical parameters). It is needed by the **adaptive** algorithm to determine when to terminate the sampling procedure. Lowering α and β and increasing +δ and -δ should increase the accuracy for the cost of more sampling.
3. In principle yes, but in practice it is difficult to replicate as the algorithm is sequential and may continue sampling (or terminate earlier).
Here is a relevant section from Uppaal SMC Tutorial (from [Documentation](https://uppaal.org/documentation/))
![image](https://github.com/UPPAALModelChecker/UPPAAL-Meta/assets/4868894/607c45f1-ea80-4409-8b90-da84310eb02e)

**Comment by fqss0436** [comment]
For question2, could you explain me more about how  hypothesis testing parameters(+δ,-δ,α,β) influence the property verification result?

I  guess the prove logic is following:

If the experimental result  k/n>=p0,(k/n means every verification runs ,for example ,17244/117236 runs  ) it is judged that the property is satisfied, if k/n<=p1, it is judged that the property is not satisfied, if k/n is between (p1,p0), it is judged according to the 
distance between k/n and p0 and p1, if the k/n is closer to p0, it is judged that 
the H0 is satisfied, otherwise it is judged that the H0 is not satisfied.

I have implemented many verifications(getting many couples of k/n) about the same  hypothesis testing property and parameters, finding they are comply with my guess as above.But I don't want to accept my guesss because it has nothing about  hypothesis testing parameters (α,β) and it shouldn't be .
